tdrs-backend/tdpservice/parsers/duplicate_manager.py

```diff
@@ -0,0 +1,193 @@
+                        schema=schema,
+                        record=record,
+                        field=None,
+                        error_message=err_msg,
+                    )
+            if is_new_max_precedence:
+                self.manager_error_dict[self.my_hash] = [error]
+            else:
+                self.manager_error_dict.setdefault(self.my_hash, []).append(error)
+            self.num_errors = len(self.manager_error_dict[self.my_hash])
+
+    def __get_partial_hash(self, record):
+        partial_hash = None
+        if record.RecordType in {"T1", "T4"}:
+            partial_hash = hash(record.RecordType + str(record.RPT_MONTH_YEAR) + record.CASE_NUMBER)
+        elif record.RecordType in {"T2", "T3", "T5"}:
+            partial_hash = hash(record.RecordType + str(record.RPT_MONTH_YEAR) + record.CASE_NUMBER +
+                                str(record.FAMILY_AFFILIATION) + record.DATE_OF_BIRTH + record.SSN)
+        return partial_hash
+
+    def add_case_member(self, record, schema, line, line_number):
+        """Add case member and generate errors if needed.
+
+        @param record: a Django model representing a datafile record
+        @param schema: the schema from which the record was created
+        @param line: the raw string line representing the record
+        @param line_number: the line number the record was generated from in the datafile
+        @return: the number of duplicate errors
+        """
+        if self.current_line_number is None or self.current_line_number != line_number:
+            self.current_line_number = line_number
+            self.record_ids.setdefault(schema.document, []).append(record.id)
+            is_exact_dup = False
+            err_msg = None
+            has_precedence = False
+            is_new_max_precedence = False
+
+            line_hash = hash(line)
+            if line_hash in self.record_hashes:
+                has_precedence, is_new_max_precedence = self.error_precedence.has_precedence(ErrorLevel.DUPLICATE)
+                existing_record_line_number = self.record_hashes[line_hash]
+                err_msg = (f"Duplicate record detected with record type "
+                           f"{record.RecordType} at line {line_number}. Record is a duplicate of the record at "
+                           f"line number {existing_record_line_number}.")
+                is_exact_dup = True
+
+            skip_partial = schema.should_skip_partial_dup
+            partial_hash = self.__get_partial_hash(record)
+            if not skip_partial and not is_exact_dup and partial_hash in self.partial_hashes:
+                has_precedence, is_new_max_precedence = self.error_precedence.has_precedence(
+                    ErrorLevel.PARTIAL_DUPLICATE)
+                existing_record_line_number = self.partial_hashes[partial_hash]
+                err_msg = (f"Partial duplicate record detected with record type "
```

Comment:
@reitermb and @victoriaatraft could I get some UX input on the structure and content of the new error message?

Reply:
partial duplicate

Reply:
We can explicitly identify which fields identify it being a partial duplicate.

Reply:
Eric to provide mockups of these messages to miles/victoria to iterate on